server/comms/rpc/rpctypes.go

```diff
@@ -0,0 +1,381 @@
+// This code is available on the terms of the project LICENSE.md file,
+// also available online at https://blueoakcouncil.org/license/1.0.0.
+
+// The specifications for the Serialize methods can be found at
+// https://github.com/buck54321/dcrdex/blob/p2sh-swap/spec/README.mediawiki#Match_negotiation
+// DRAFT NOTE: Update this link ASAP.
+
+package rpc
+
+import (
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+)
+
+// Error codes
+const (
+	RPCParseError = iota
+	RPCUnknownMethod
+	RPCInternal
+	RPCQuarantineClient
+	RPCVersionUnsupported
+	RPCUnknownMatch
+	RPCInternalError
+	SignatureError
+	SerializationError
+	TransactionUndiscovered
+	ContractError
+	SettlementSequenceError
+	ResultLengthError
+	IDMismatchError
+	RedemptionError
+	IDTypeError
+	AckCountError
+)
+
+const (
+	JSONRPCVersion    = "2.0"
+	MatchMethod       = "match"
+	InitMethod        = "init"
+	AuditMethod       = "audit"
+	RedeemMethod      = "redeem"
+	RedemptionMethod  = "redemption"
+	RevokeMatchMethod = "revoke_match"
+)
+
+// Signable allows for serialization and signing.
+type Signable interface {
+	Serialize() ([]byte, error)
+	SetSig([]byte)
+	SigBytes() []byte
+}
+
+// signable implements Signable, and should be embedded by any rpc type the
+// is serializable and needs to encode a Sig field.
+type signable struct {
+	Sig string `json:"sig"`
```

Comment:
Alternatively, could store as bytes via a custom type implementing `json.Mashaller` and `json.Unmarshaller`.

Reply:
See https://github.com/decred/dcrdex/blob/d3af131c00615d8bfd7b7566f30d0480f16cecbe/server/comms/rpc/rpctypes.go#L49

Reply:
I actually meant to use `Bytes` here.